mavros/src/plugins/setpoint_position.cpp

```diff
@@ -93,12 +100,25 @@ class SetpointPositionPlugin : public plugin::PluginBase,
 		 */
 		const uint16_t ignore_all_except_xyz_y = (1 << 11) | (7 << 6) | (7 << 3);
 
-		auto p = ftf::transform_frame_enu_ned(Eigen::Vector3d(tr.translation()));
-		auto q = ftf::transform_orientation_enu_ned(
-					ftf::transform_orientation_baselink_aircraft(Eigen::Quaterniond(tr.rotation())));
+		auto p = [&] {
+			if (static_cast<MAV_FRAME>(mav_frame) == MAV_FRAME::BODY_NED || static_cast<MAV_FRAME>(mav_frame) == MAV_FRAME::BODY_OFFSET_NED) {
+				return ftf::transform_frame_baselink_aircraft(Eigen::Vector3d(tr.translation()));
+			} else {
+				return ftf::transform_frame_enu_ned(Eigen::Vector3d(tr.translation()));
+			}
+		} ();
+
+		auto q = [&] {
+			if (static_cast<MAV_FRAME>(mav_frame) == MAV_FRAME::BODY_NED || static_cast<MAV_FRAME>(mav_frame) == MAV_FRAME::BODY_OFFSET_NED) {
```

Comment:
With correct type only what you need: `mav_frame == MAV_FRAME::BODY_NED || ...`

Reply:
done